src/record/api.rs

```diff
@@ -36,7 +36,34 @@ macro_rules! nyi {
 
 /// Row API to represent a nested Parquet record.
 #[derive(Clone, Debug, PartialEq)]
-pub enum Row {
+pub struct Row {
+  fields: Vec<(String, RowField)>
+}
+
+pub fn make_row(fields: Vec<(String, RowField)>) -> Row {
```

Comment:
Can you add some doc to the function? Thanks.
Would it be better to keep it as macro, since it is used in different places? Would it be inlined? If so, then it is fine.